### PR TITLE
[6736] Form freezes on save if user cancels a policy suggestion dialog

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/6736

### CODE REVIEW

File: static-assets/components/cstudio-common/common-api.js [hunk 1]
```diff
@@ -6224,12 +6224,11 @@ var nodeOpen = false,
         $(document).on('click', `.notifyjs-${styleName}-${id} .yes`, onOk);
       },
 
-      showConfirmDialog: function (title, body, callback, okButtonText, cancelButtonText) {
-        const onOk = 'confirmDialogOnOk';
-        const onCancel = 'confirmDialogOnCancel';
-        let unsubscribe, cancelUnsubscribe;
+      showConfirmDialog: function ({ title, body, onOk, onCancel, okButtonText, cancelButtonText }) {
```
Comment (rart) [inline static-assets/components/cstudio-common/common-api.js]: Make this change backward compatible, add to changelog and test all affected usages
```suggestion
      showConfirmDialog: function (
        // title, body, callback, okButtonText, cancelButtonText
        // { title, body, onOk, onCancel, okButtonText, cancelButtonText }
      ) {
        const { title, body, onOk, onCancel, okButtonText, cancelButtonText } = typeof arguments[0] === 'object' ? arguments[0] : { title: arguments[a], body: arguments[b], etc... }
```

File: static-assets/components/cstudio-common/common-api.js [hunk 2]
```diff
@@ -6258,20 +6257,20 @@ var nodeOpen = false,
...
             }
           });
 
-          unsubscribe = CrafterCMSNext.createLegacyCallbackListener(onOk, () => {
-            callback();
-            cancelUnsubscribe();
-          });
-          cancelUnsubscribe = CrafterCMSNext.createLegacyCallbackListener(onCancel, () => {
-            unsubscribe();
+          unsubscribe = CrafterCMSNext.createLegacyCallbackListener(confirmDialogEvent, ({ type }) => {
```
Comment (rart) [inline static-assets/components/cstudio-common/common-api.js]: We could/should replace this with utils.dom.createCustomEventBlahblah, as long as keeps working the same externally, of course.

File: static-assets/components/cstudio-forms/forms-engine.js [hunk 3]
```diff
@@ -1680,21 +1682,26 @@ var CStudioForms =
...
+                      CStudioAuthoring.Utils.showConfirmDialog({
+                        body: formatMessage(formEngineMessages.createPolicy, {
+                          originalPath: entityId,
+                          path: modifiedValue
+                        }),
+                        onOk: () => {
                           saveContent();
+                        },
+                        onCancel: () => {
+                          setButtonsEnabled(true);
```
Comment (rart) [inline static-assets/components/cstudio-forms/forms-engine.js:1694]: Where are these getting disabled?

**Review by rart — CHANGES_REQUESTED** [review]

File: static-assets/components/cstudio-forms/forms-engine.js (hunk 3 above)
Comment (jvega190) [reply static-assets/components/cstudio-forms/forms-engine.js:1694]: They're getting disabled at the beginning of the save function (right when clicking on save button)

File: ui/app/src/CHANGELOG.md [hunk 4]
```diff
@@ -37,6 +37,10 @@
   * `usePreviewUrlControl`: Removed `history` prop. Retrieval of search and navigate (previously called 'push') is now done internally.
 * Removed deprecated `aws-file-upload` and `aws-video` upload controls.
 
+## 4.1.5
+* [common-api.js]
+  * `CStudioAuthoring.Utils.showConfirmDialog`: Updated arguments to support an object with the following properties instead of individual arguments: `title`, `message`, `onOk`, `onCancel`, `okButtonText`, `cancelButtonText`. Also, made it backward compatible (with multiple arguments).
```
Comment (rart) [inline ui/app/src/CHANGELOG.md]:
```suggestion
  * `CStudioAuthoring.Utils.showConfirmDialog`: Added function overload to receive a `props` style object as first and only argument. The props argument would contain all ConfirmDialog props. Original set of arguments still supported for backward compatibility.
```

File: static-assets/components/cstudio-common/common-api.js [hunk 5]
```diff
@@ -6224,12 +6224,24 @@ var nodeOpen = false,
...
+            : {
+                title: arguments[0],
+                body: arguments[1],
+                onOk: arguments[2],
+                okButtonText: arguments[3],
+                cancelButtonText: arguments[4]
+              };
 
-        if (callback) {
+        const confirmDialogEvent = 'confirmDialogEvent';
```
Comment (rart) [inline static-assets/components/cstudio-common/common-api.js]:
```suggestion
        const confirmDialogEvent = 'commonAPIConfirmDialogEvent';
```

**Review by rart — CHANGES_REQUESTED** [review]

**Review by rart — APPROVED** [review]

**Pull request merged**